core/src/subgraph/runner.rs

```diff
@@ -144,12 +197,17 @@ where
 
             let block_stream_canceler = CancelGuard::new();
             let block_stream_cancel_handle = block_stream_canceler.handle();
+            // TriggerFilter needs to be rebuilt eveytime the blockstream is restarted
```

Comment:
Exactly this, if the build returns the trigger filter and replace it then we know what mutations it needs to make and the build_filter function can then take a reference instead of mutable ref

Reply:
Thanks, just did that